Remember selected panels and handle case when user hasn't logged in

n-riesco opened the pull request:
This PR implements:

* The selected panel for each connection is stored in the component state and new connections default to the Connection panel.

* If the users hasn't logged into Plotly, the Plot.ly panel shows a link to the login page.

tarzzz commented on app/components/Settings/Settings.react.js:
```diff
@@ -440,6 +462,12 @@ class Settings extends Component {
+                                    ): (
+                                        <div>
+                                            <p>
+                                                <a onClick={() => window.location.assign('/login')}>Log into Plotly</a>
```
The message should be a little more informing imho:

> Log in to Plotly to generate unique SSL certificates, schedule and save queries.

@chriddyp thoughts?

chriddyp replied:
yeah, that sounds good to me

tarzzz commented on app/components/Settings/Settings.react.js:
```diff
@@ -42,6 +43,27 @@ class Settings extends Component {
+            });
+        };
+
+        checkIfLoggedIn();
```
@n-riesco this wont work for electron app unfortunately.. Since electron app doesn't require log-in to access settings (or any url for that matter) .. I would suggest checking for `username`, which is stored in its parent component..

tarzzz replied:
We need to send this state: https://github.com/plotly/falcon-sql-client/blob/master/app/components/Configuration.react.js#L26
as a prop to `Settings` and take use it as a measure of user being logged-in or not. Let me know if you need more info on this.. 😸

n-riesco replied:
@tarzzz Not sure I understand, it does work on my machine.

tarzzz replied:
@chriddyp This should work for electron-app and we are letting server/headless version to handle their own certificates.. so should be good?

chriddyp replied:
@tarzzz - this logic makes sense to me for the headless version however it won't apply to the multi-user server/headless version - I'm guessing that a check for `username` will work in that case

n-riesco commented:
@tarzzz @chriddyp I've just tested master and `username` isn't working in the desktop or the web app (see the screen captures below). I'll update this PR once this issue with login is fixed.

![peek 2017-10-20 20-55](https://user-images.githubusercontent.com/6199391/31840086-ac542492-b5db-11e7-8f53-5dea36250cc5.gif)

![peek 2017-10-20 21-08](https://user-images.githubusercontent.com/6199391/31840092-b1e6b370-b5db-11e7-9c49-1d486ab18880.gif)

tarzzz commented:
@n-riesco For server/headless version, you need to add the following lines to your `settings.yaml`:

```
ALLOWED_USERS:
    - n-riesco
```

The issue in electron-app is a known issue, but I have been unable to replicate it on my machine so far. Adding those lines to settings.yaml would let you authenticate on server-app and should be good for your dev. for this PR..

n-riesco commented:
@tarzzz I've added those lines to `settings.yaml` and fix the auth error, but I get stuck on the login page. And if I reload the `localhost:9494`, I only see the link to login again.

See the screen capture:

![peek 2017-10-21 19-52](https://user-images.githubusercontent.com/6199391/31855010-f0d0b536-b699-11e7-9641-0db849bf9b18.gif)

n-riesco commented:
@tarzzz @chriddyp I've updated the PR:
- I've fixed the `loadURL` issue I showed above, so that now oauth2 works in the desktop app.
- I've changed the approach to determine whether the user is logged (now it uses `username`).
- I've also tried to address some of @chriddyp 's feedback in https://github.com/plotly/streambed/issues/10160#issuecomment-337425410 (moved the login/logout link in the desktop app into the Plot.ly panel)

Oauth2 isn't working yet in the web app (even after setting ALLOWED_USERS). Do you want me to have a look at this issue?

tarzzz commented:
@n-riesco I suspect the Oauth issue is related to `SSL_ENABLED` setting.

When `SSL_ENABLED=true` in `settings.yaml`, `secure` flag is set on cookies, and they are not sent over `http` connections.

tarzzz commented on app/components/Configuration.react.js:
```diff
@@ -77,24 +77,15 @@ class Configuration extends Component {
             <div className="fullApp">
-                {loginMessage}
-                <Settings/>
+                <Settings username={this.state.username} logout={this.logOut}/>
```
👍

tarzzz commented on backend/main.development.js:
```diff
@@ -39,6 +39,7 @@ app.on('ready', () => {
      * This allows us to send pass information to electron renderer from
      * inside the server.
      */
+    server.electronWindow = mainWindow;
```
😸

tarzzz commented on app/components/Settings/Settings.react.js:
```diff
@@ -461,11 +444,18 @@ class Settings extends Component {
                                             <p>
                                                 <a onClick={() => window.location.assign('/login')}>Log into Plotly</a>
+                                                <br/>
+                                                Please, login in order to connect your queries to your Plotly account.
```
Oh, didn't saw this line earlier. I thought the message only said "Login to Plotly". .. No change required here imo..

n-riesco commented:
Closing since all the commits in this PR have been cherry-picked into #206.
